### ISSUE TITLE
[Bug] "User interface scaling" Auto setting does not use "Double" for 4K monitor

### ISSUE THREAD

**Issue by sanecito** [linuxmint/Cinnamon#7736]
```
 * Cinnamon version (cinnamon --version): 3.8.7
 * Distribution - Mint 19
 * Graphics hardware *and* driver used: Advanced Micro Devices, Inc. [AMD/ATI] Hawaii PRO [Radeon R9 290/390] / "Kernel driver in use: radeon"
 * 64 bit
 ```

**Issue**
When using the 'Auto' setting found in General > Desktop Scaling it does not use 'Double' on a system with a 4K monitor set as primary; it seems to use 'Normal' which makes icons difficult to read/click.

**Steps to reproduce**
1. Use a 4K monitor set as primary
2. Set "User interface scaling" to 'Auto'
3. Note the scaling level (i.e. is it the scaling value of 'Normal' thus making it hard to use UI elements?)

**Expected behaviour**
'Auto' value should use the 'Double' scaling value when a 4K monitor is set as primary, not the 'Normal' value

**Other information**
Using a dual monitor setup; a 22" 1080p and a 27" 4K as primary. Not sure if this would have any bearing, but FYI.

**Comment by JordanL2** [comment]
I can confirm this happens with:
* two 4k 28'' monitors
* connected via Display Port
* using a NVidia GTX 980Ti with the binary driver

Even with this screen size, at 4k it really needs the double scaling.

**Comment by icarter09** [comment]
@sanecito or @JordanL2: either of you still experiencing this issue?

**Comment by JordanL2** [comment]
I haven't been using Cinnamon recently so I can't retest.

**Comment by icarter09** [comment]
@JordanL2 no worries and thanks for the update.

**Comment by sanecito** [comment]
Same, but will test with a live stick this weekend to see if issue persists.

**Comment by sanecito** [comment]
Different monitor setup (laptop), but I can confirm on the following monitor running a 19.2 Cinnamon Linx Mint USB which should definitely need double, Auto doesn't not use Double.
```

2560x1400 pixel (677x381 mm)
96x96 dots per inch
```

**Comment by mtwebster** [comment]
https://github.com/linuxmint/cinnamon-settings-daemon/blob/master/plugins/xsettings/csd-xsettings-manager.c#L227-L230

If you halve this, you're only 1280x700 pixels - the height extremely restrictive, and many apps would not fit the screen.  I would argue that even 1500 as the threshold for autoset is also too small, but it's a bit subjective, which is why you can override it.

I'm not sure about your original setup above, you didn't actually report the 4k monitor's size and resolution.  Both play a factor in the calculations (which unfortunately can also depend on how well the monitor reports itself to the system).

**Comment by sanecito** [comment]
Valid point, I'll update with the specs from the original post within 12 hours when I should have access to the setup.

**Comment by sanecito** [comment]
It's a ViewSonic XG2700-4K 27" 3840 x 2160(4K) Ultra HD, running at (on Manjaro):

```
  dimensions:    3840x2160 pixels (965x271 millimeters)
  resolution:    101x101 dots per inch
```
For whatever reason at least with the live USB, running through my 2080 Ti it can't detect my 4K monitor over DP using either the noveau, nvidia 435, or nvidia430 drivers (detects the 1080p monitor just fine and amusingly labels it 'laptop'). So unfortunately I don't seem to be in a position to easily help confirm this is now resolved. If you want to close it out because the logic for using 'Double' in Auto under certain criteria has improved since 2018/07/14, that's fine by me as my use cases demands the use of an Arch based distro for easily getting the latest drivers so I don't plan to jump back to Mint.

If I'm remembering things correctly, what probably happened when I filed this issue was that Linux Mint (and Manjaro still too, although I found can be fixed with xrandr manually) only has global scaling, so I likely set the 4K to 1080p, which then didn't meet the 1500 min height trigger threshold.

So in short, real issue at the time (and still exists) for me is that Linux Mint doesn't have independent scaling for multi-monitor setups, at least in the GUI from what I can tell.

However, it still doesn't explain @JordanL2's issue of 2x 4K monitors:

> I can confirm this happens with:
> 
>     two 4k 28'' monitors
>     connected via Display Port
>     using a NVidia GTX 980Ti with the binary driver
> 
> Even with this screen size, at 4k it really needs the double scaling.

**Comment by leigh123linux** [comment]
Auto scaling has never worked for me either.

```
screen #0:
  dimensions:    3840x2160 pixels (1016x572 millimeters)
  resolution:    96x96 dots per inch
  depths (7):    24, 1, 4, 8, 15, 16, 32
```
